src/greynoise/api/__init__.py

```diff
@@ -112,6 +109,21 @@ def __init__(
         self.integration_name = integration_name
         self.session = requests.Session()
 
+        if cache_ttl is None:
```

Comment:
we are overloading the `__init__` function. Instead, the `__init__` function should call on a `create_cache` (or a more appropriate named) function to create the cache. This will simplify how much goes on inside the `__init__`. A more important point is this code is not defensive. For example, using this current setup if I pass "foo" and "bar" as `cache_max_size` and `cache_ttl` this will generate a runtime exception that we are not handling. Instead, we should have the function responsible of creating the cache, to first inspect the parameters submitted before trying to create the cache. It should return `None` if it cant create the cache or the actual cache. This allows the caller to have a simple logic when trying to create a cache.

```python
cache = getdaCache(max_size="foo", ttl="bar") 
if not cache:
     return "we could not make the cache"
...
```



**exception reference**
Note the exception shows up when you try to insert an item, not when you try to create a cache

```python
>>> import cachetools
>>> t = cachetools.TTLCache(maxsize="a")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() missing 1 required positional argument: 'ttl'
>>> t = cachetools.TTLCache(maxsize="foo", ttl="bar")
>>> t.setdefault("rockets", "foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/gmo/.pyenv/versions/3.7.0/lib/python3.7/site-packages/cachetools/ttl.py", line 189, in setdefault
    return Cache.setdefault(self, *args, **kwargs)
  File "/Users/gmo/.pyenv/versions/3.7.0/lib/python3.7/site-packages/cachetools/abc.py", line 42, in setdefault
    self[key] = value = default
  File "/Users/gmo/.pyenv/versions/3.7.0/lib/python3.7/site-packages/cachetools/ttl.py", line 89, in __setitem__
    cache_setitem(self, key, value)
  File "/Users/gmo/.pyenv/versions/3.7.0/lib/python3.7/site-packages/cachetools/cache.py", line 46, in __setitem__
    if size > maxsize:
TypeError: '>' not supported between instances of 'int' and 'str'
```

Reply:
@guillermo-menjivar : see the latest commit, is this what you were asking?